test/integration/externalnode_test.go

```diff
@@ -0,0 +1,361 @@
+// Copyright 2022 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"math/rand"
+	"path"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	antreav1alpha1 "antrea.io/antrea/pkg/apis/crd/v1alpha1"
+	antreav1alpha2 "antrea.io/antrea/pkg/apis/crd/v1alpha2"
+	"antrea.io/nephe/apis/crd/v1alpha1"
+	"antrea.io/nephe/pkg/controllers/config"
+	"antrea.io/nephe/test/utils"
+)
+
+var _ = Describe(fmt.Sprintf("%s,%s: ExternalNode", focusAws, focusAzure), func() {
+	const (
+		namespaceName   = "external-node-test"
+		testAccountName = "external-node-test-account"
+	)
+
+	var (
+		pollInterval  uint = 30
+		namespace     *corev1.Namespace
+		account       *v1alpha1.CloudProviderAccount
+		selector      *v1alpha1.CloudEntitySelector
+		secret        *corev1.Secret
+		cloudProvider string
+		getEnCmd      string
+		getEeCmd      string
+		vmIdToMatch   string
+		testVm        string
+		nic           string
+		publicIp      string
+		privateIp     string
+	)
+
+	createNS := func() {
+		logf.Log.Info("Create", "namespace", namespace.Name)
+		err := k8sClient.Create(context.TODO(), namespace)
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	deleteNS := func() {
+		logf.Log.Info("Delete", "namespace", namespace.Name)
+		err := k8sClient.Delete(context.TODO(), namespace)
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	createAccount := func() {
+		logf.Log.Info("Create", "account", account.Name)
+		err := k8sClient.Create(context.TODO(), secret)
+		Expect(err).ToNot(HaveOccurred())
+		err = k8sClient.Create(context.TODO(), account)
+		Expect(err).ToNot(HaveOccurred())
+		time.Sleep(5 * time.Second)
+	}
+
+	deleteAccount := func() {
+		logf.Log.Info("Delete", "account", account.Name)
+		err := k8sClient.Delete(context.TODO(), account)
+		Expect(err).ToNot(HaveOccurred())
+		err = k8sClient.Delete(context.TODO(), secret)
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	applyCloudEntitySelector := func(selectorExists bool, agented bool) {
```

Comment:
Also add the test coverage for other match like VPC ID.

Reply:
We already have entity selector test covering VPC ID testing match selector, I thought the focus of this test is to test external node lifecycle, hence i didn't want to add different match criteria in CES.
If you think we need to have VPC ID in this test too, i will take up soon after the ongoing task.

Reply:
Yes, we need test for all combination, VPC ID, VM ID, VMName and VPC Name. As the External Node Cache indexer is different from External Entity, we cannot rely on those tests. you can take it up next